File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/postgresql/EnumType.scala
package dev.typr.foundations.docs.postgresql
import dev.typr.scalafoundations.*
import dev.typr.scalafoundations.data.*



@SuppressWarnings(Array("unused"))
object EnumType:
  //start
  // Define your Scala 3 enum
  enum Status:
    case PENDING, ACTIVE, COMPLETED

  // Create a PgType using bimap (Scala 3 enums don't extend java.lang.Enum)
  val statusType: PgType[Status] = PgTypes.text.bimap(
    Status.valueOf,
    _.toString
  )
  //stop

File: documentation-examples-java/src/java/dev/typr/foundations/docs/postgresql/EnumType.java
package dev.typr.foundations.docs.postgresql;

import dev.typr.foundations.PgType;
import dev.typr.foundations.PgTypes;

@SuppressWarnings("unused")
public class EnumType {
  // start
  // Define your Java enum
  public enum Status {
    PENDING,
    ACTIVE,
    COMPLETED
  }

  // Create a PgType — pass values(), no reflection
  PgType<Status> statusType = PgTypes.ofEnum("status", Status.values());
  // stop
}
